src/proc/hdr-merge.cpp

```diff
@@ -112,6 +124,8 @@ namespace librealsense
                 (merged_d_profile.height() != new_d_profile.height()))
```

Comment:
Split the condition so that the counter reset will occur only if the frame index is restarted

Reply:
This is performed during the process method, which is not accessed at all when the metadata is not available - so there is no way the counter can be reset here.
The solution would be to have a member of the frame counter and save it at each frame. When the input frame counter is less than the saved one, this will be a pipe restart detection and the counter will be reset - do you agree?

Reply:
This is not possible either since the frame counter is not available when metadata is not available - TDB